connection_params.ts

```diff
@@ -78,20 +81,26 @@ export class ConnectionParams {
       config = dsn;
     }
 
-    this.database = selectFrom([config, pgEnv], "database");
-    this.host = selectFrom([config, pgEnv, DEFAULT_CONNECTION_PARAMS], "host");
-    this.port = selectFrom([config, pgEnv, DEFAULT_CONNECTION_PARAMS], "port");
-    this.user = selectFrom([config, pgEnv], "user");
+    let potentiallyNull: { [K in keyof IConnectionParams]?: string } = {
+      database: selectFrom([config, pgEnv], "database"),
+      user: selectFrom([config, pgEnv], "user")
+    };
+
+    this.host =
+      selectFrom([config, pgEnv], "host") || DEFAULT_CONNECTION_PARAMS["host"];
```

Comment:
@denniscollective can you move `DEFAULT_CONNECTION_PARAMS["host"]` into array inside `selectFrom`?

Reply:
if we do that we need to have a bunch more ! type assertions at the top, because it could be a string | undefined. I was trying to avoid those as much as possible. I thought about making another function that was ```selectWithDefault``` but that  seemed like overkill.

Reply:
> I thought about making another function that was `selectWithDefault` but that seemed like overkill.

:+1: if it makes code more readable then I'm okay with that